core/src/main/scala/kafka/log/Log.scala

```diff
@@ -322,6 +327,11 @@ class Log(@volatile private var _dir: File,
     // deletion.
     producerStateManager.removeStraySnapshots(segments.values().asScala.map(_.baseOffset).toSeq)
     loadProducerState(logEndOffset, reloadFromCleanShutdown = hadCleanShutdown)
+
+    // Recover topic ID if present
+    if (!partitionMetadataFile.get.isEmpty()) {
+      topicId = partitionMetadataFile.get.read().topicId
+    }
```

Comment:
Seems neater to use `partitionMetadataFile.foreach` instead of using `.get` twice.